Automattic/wp-calypso
# Account: use Redux to change user password

…instead of `lib/user-settings`

#### Changes proposed in this Pull Request

* Remove `userSettings` prop from password route controller
* Refactor `AccountPassword` component to use Redux to change password

#### Testing instructions

* Go to `/me/security/password`
* Attempt to change your password (try scenarios like failing password validation and request blocking)
* Is the behaviour the same as on prod?

After you changed your password successfully the browser window should reload and ask you for a new 2fa token in case you have it set up.

part of #24162
